src/codemagic/apple/resources/resource.py

```diff
@@ -72,7 +72,7 @@ def __post_init__(self):
 class Links(DictSerializable):
     _OMIT_IF_NONE_KEYS = ('related',)
 
-    def __init__(_self, self: str, related: Optional[str] = None):
+    def __init__(_self, self: Optional[str] = None, related: Optional[str] = None):
```

Comment:
The change is required because few relationships are not specified in API, and they don't have `self` field, e.g.

```
            "perfPowerMetrics":{
               "links":{
                  "related":"https://api.appstoreconnect.apple.com/v1/builds/807b1541-9601-48da-8169-9130f93d4b90/perfPowerMetrics"
               }
```

Reply:
fixed [here](https://github.com/codemagic-ci-cd/cli-tools/pull/67/commits/71f2ca10806aa46be85d0feca84182780939f3d7#diff-b736b487114b15d1320d4be383e9cb605e79ac0e683d6949d012fc6215dcf44eR74)